Applications/src/detect-cardiac-phases.cc

```diff
@@ -92,31 +98,30 @@ int main(int argc, char* argv[] )
     for (int k = 0; k < cine.GetZ(); ++k)
     for (int j = 0; j < cine.GetY(); ++j)
     for (int i = 0; i < cine.GetX(); ++i) {
-      dif = (blurred.GetAsDouble(i,j,k,t) - blurred.GetAsDouble(i,j,k,0)) / cinedis;
-      similarity[t] += dif*dif;
+      similarity[t] += pow(double((blurred(i,j,k,t) - blurred(i,j,k,0))) / cinedis, 2);
     }
     if (verbose) {
       cout << "similarity : " << similarity[t] << endl;
     }
   }
-  for (i = 0; i < frames; i++) {
+  for (int i = 0; i < frames; i++) {
     similarity[i] = sqrt(similarity[i]);
   }
   // Smooth similarity
-  for (i = 1; i < frames - 1; i++) {
+  for (int i = 1; i < frames - 1; i++) {
     smoothsimilarity[i] = (similarity[i-1] + similarity[i] + similarity[i+1]) / 3;
   }
   // Find min similarity
   dif = 0;
```

Comment:
That's a bit picky, but could you remove the declaration of this variable earlier on and just declare it here?

```
double dif = 0;
// code that actually uses dif
```